### PR TITLE
Auth context

### PR DESCRIPTION
**Pull request by berenteb**
Holy Moly this is a big boy.
What I have created (as a concept):
- AuthContext to get the isLoggedIn state from everywhere inside
- useAuthContext hook to simplify usage
- loginRequired attribute to Page (optional) -> display UnauthorizedPage when not logged in
- loginRequired attribute to navItems (optional) -> appear on navbar only when logged in
- AuthButton -> Login/logout based on loggedIn state
- This only works with localStorage now, ATM. AuthSch is not working, good for testing anyways. Will have to change localstorage to cookie.get()...
- Added loginRequired attribute to Profile and Achievement pages

### CODE REVIEW

File: src/main/client/package.json [hunk 1]
```diff
@@ -36,8 +36,10 @@
     "@emotion/styled": "11",
     "@fontsource/commissioner": "4.5.0",
     "@fontsource/raleway": "4.5.0",
+    "@types/js-cookie": "^3.0.1",
```
Comment (triszt4n) [inline src/main/client/package.json:39]: this should be dev dependency

**Review by triszt4n — COMMENTED** [review]
Vannak bajok a rebase-eléssel és mergeléssel, majd ha sehogy sem sikerül megoldani, szóljál

**Comment by berenteb** [comment]
Pedig master-ra van reabselve 🗡️

**Comment by triszt4n** [comment]
De van benne egy random merge

image

Gyere DC 15 perc múlva

**Pull request merged**